contrib/python/podman/libs/images.py

```diff
@@ -135,3 +128,7 @@ def search(self, id, limit=25):
             results = podman.SearchImage(id)
         for img in results['images']:
             yield img
+
+    def get(self, id):
+        """Get Image from id."""
+        return next(iter([i for i in self.list() if i.id == id] or []), None)
```

Comment:
This iterates through the whole list even after finding a match.  Since `Images.list` is already `yield` based, you should be able to use:

```python
return next((i for i in self.list() if i.id == id), None)
```

And you can confirm the difference with:

```pycon
>>> next((a for a in [1,2,3] if print('check {}'.format(a)) is None and a > 3), None)
check 1
check 2
check 3
>>> next((a for a in [1,2,3] if print('check {}'.format(a)) is None and a > 1), None)
check 1
check 2
2
>>> next(iter([a for a in [1,2,3] if print('check {}'.format(a)) is None and a > 1]), None)
check 1
check 2
check 3
2
```

Reply:
I went ahead and made this change.  We're going to move this functionality onto the service in time, so I wasn't too worried.